src/tools/simeng/gdb/GDBStub.cc

```diff
@@ -0,0 +1,375 @@
+          std::cout << "[SimEng:GDBStub] <- Received packet: " << packet
+                    << std::endl;
+        else
+          std::cout << RED << "[SimEng:GDBStub] <- Unknown message"
+                    << std::endl;  // TODO: take action
+        std::cout << RESET;
+      }
+
+      std::string response = "";
+      if (!noAckMode_) response += "+";  // acknowledgement
+
+      std::regex qSupported_regex("^qSupported:(.*)");
+      std::smatch qSupported_match;
+
+      std::string checksum = packet_match[2].str();
+      std::string expected = computeChecksum(packet);
+
+      if (checksum != expected) {
+        if (verbose_)
+          std::cout << RED << "[SimEng:GDBStub]    Checksum failed\n"
+                    << "   Expected checksum: " << expected
+                    << "   , but received: " << checksum << std::endl
+                    << RESET;
+        response = "-";            // request for re-transmission
+      } else if (packet == "?") {  // reason for halting
+        response += generateReply(BREAK_POINT_HIT);
+      } else if (packet == "s") {  // step one instruction
+        core_.tick();
+        dataMemory_.tick();
+        response += generateReply(BREAK_POINT_HIT);
+      } else if (packet == "c") {  // continue until next breakpoint
+        std::string message = handleContinue();
+        response += generateReply(message);
+      } else if (packet == "g") {  // read registers
+        std::string registers = handleReadRegisters();
+        response += generateReply(registers);
+      } else if (packet[0] == 'p') {  // read single register
+        std::regex reg_regex("^p([0-9a-f]*)");
+        std::smatch reg_match;
+        regex_match(packet, reg_match, reg_regex);
+        std::string registerValue = handleReadSingleRegister(reg_match[1]);
+        response += generateReply(registerValue);
+      } else if (packet[0] == 'm') {  // read memory
+        std::regex mem_regex("^m([0-9a-f]*),([0-9a-f]*)");
+        std::smatch mem_match;
+        regex_match(packet, mem_match, mem_regex);
+        std::string memoryValue = handleReadMemory(mem_match[1], mem_match[2]);
+        response += generateReply(memoryValue);
+      } else if (packet[0] == 'Z' ||
+                 packet[0] == 'z') {  // create/remove breakpoint
+        std::regex break_regex(
+            "^([zZ])([0-4]),([0-9a-f]+),([0-9a-f]+)");  // format:
+                                                        // z/Z,type,address,kind
+        std::smatch break_match;
+        regex_match(packet, break_match, break_regex);
+        std::string type = break_match[2];
+        std::string address = break_match[3];
+        // std::string kind = break_match[4]; // kind not used
+        if (packet[0] == 'Z')
+          handleCreateBreakpoint(type, address);
+        else
+          handleRemoveBreakpoint(type, address);
+        response += generateReply(GDB_OK);
+      } else if (packet[0] == 'M') {  // write memory
```

Comment:
Request from PR author, "remove the lines in GDBStub.cc that check for 'M' and 'G' due to lacking support".